### PR TITLE
fix: Add `WinUIUnhandledExceptionIntegration` on Windows only

### PR DESCRIPTION
**Pull request by bitsandfoxes**
Fixes https://github.com/getsentry/sentry-dotnet/issues/3050

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,10 @@ If you have conflicts, you can opt-out by adding the following to your `csproj`:
 </PropertyGroup>
 ```
 
+### Fixes
+
+- The SDK no longer adds the `WinUIUnhandledExceptionIntegration` on non Windows platforms ([#3043](https://github.com/getsentry/sentry-dotnet/pull/3043))
```
Comment (bitsandfoxes) [inline CHANGELOG.md:17]:
```suggestion
- The SDK no longer adds the `WinUIUnhandledExceptionIntegration` on non Windows platforms ([#3055](https://github.com/getsentry/sentry-dotnet/pull/3055))
```

File: src/Sentry/SentryOptions.cs [hunk 2]
```diff
@@ -186,7 +186,7 @@ internal IEnumerable<ISdkIntegration> Integrations
             }
 #endif
 
-#if NET5_0_OR_GREATER && !__MOBILE__
+#if NET5_0_OR_GREATER && WINDOWS
```
Comment (bruno-garcia) [inline src/Sentry/SentryOptions.cs:189]: Does this mean building for `net8.0` for example won't have this handler and is that by design?
Sounds right to me, but checking, if a WInUI app has to always be built with `net8.0-windowsXYZ` or something.

Comment (bruno-garcia) [reply src/Sentry/SentryOptions.cs:189]: @lucas-zimerman might know

Comment (bruno-garcia) [reply src/Sentry/SentryOptions.cs:189]: Note: we don't have a `net8.0-windows` target in the SDK so I think this isn't getting what you expect

Comment (bitsandfoxes) [reply src/Sentry/SentryOptions.cs]: This is a big TIL. So we have to check during runtime whether to add it depending on the current platform.

Comment (lucas-zimerman) [reply src/Sentry/SentryOptions.cs]: to be honest I don't follow about the support of UnhandledException on .NET 8.

Comment (bruno-garcia) [reply src/Sentry/SentryOptions.cs]: > This is a big TIL. So we have to check during runtime whether to add it depending on the current platform.

yeah all the compiler flags affect the DLLs we output. And `net8.0` is platform agnostic so ANDROID/WINDOWS/etc wil not be set.

File: test/Sentry.Tests/SentryOptionsTests.verify.cs [hunk 3]
```diff
@@ -3,9 +3,12 @@ namespace Sentry.Tests;
 [UsesVerify]
 public partial class SentryOptionsTests
 {
-    [Fact]
+    [SkippableFact]
     public Task Integrations_default_ones_are_properly_registered()
     {
+        // Windows additionally adds `WinUIUnhandledExceptionIntegration`
+        Skip.If(RuntimeInformation.IsOSPlatform(OSPlatform.Windows));
```
Comment (bruno-garcia) [inline test/Sentry.Tests/SentryOptionsTests.verify.cs:10]: should we have two tests then? one that tests we have this on Windows? Instead of skipping

Comment (bitsandfoxes) [reply test/Sentry.Tests/SentryOptionsTests.verify.cs:10]: I had that but it messed with the verify. I guess I need to run this on windows then too to generate the expected file?

Comment (bruno-garcia) [reply test/Sentry.Tests/SentryOptionsTests.verify.cs:10]: yeah you need to have a file per platform, just saw this on another PR:
image

Comment (bruno-garcia) [reply test/Sentry.Tests/SentryOptionsTests.verify.cs:10]: one of those per OS I imagine

**Review by bruno-garcia — APPROVED** [review]
approving to unblock but best finding out a way to test it instead of skipping on Windows

**Comment by bruno-garcia** [comment]
> approving to unblock but best finding out a way to test it instead of skipping on Windows

@bitsandfoxes is there a ticket to track this? good to link to this PR

**Comment by bitsandfoxes** [comment]
> > approving to unblock but best finding out a way to test it instead of skipping on Windows
> 
> @bitsandfoxes is there a ticket to track this? good to link to this PR

Follor up PR is here https://github.com/getsentry/sentry-dotnet/pull/3063

**Pull request merged**